apps/clockbyhenry/clock_by_henry.star

```diff
@@ -0,0 +1,499 @@
+"""
+
+# Big-Number Clock With Date
+#
+# Copyright (c) 2022 Henry So, Jr.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+load("re.star", "re")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+
+# parameters:
+# location - the location for rendering
+# use_12h - if 12-hour time should be used
+# blink_time - if the time separator should blink
+# season_theme - the base color set to use (index of SEASON_THEMES)
+# d - the color for the time during the day
+# n - the color for the time during the night
+# s0d - the color for the date during season 0 in the day
+# s1d - the color for the date during season 1 in the day
+# s2d - the color for the date during season 2 in the day
+# s3d - the color for the date during season 3 in the day
+# s0n - the color for the date during season 0 in the night
+# s1n - the color for the date during season 1 in the night
+# s2n - the color for the date during season 2 in the night
+# s3n - the color for the date during season 3 in the night
+def main(config):
+    location = config.get(P_LOCATION)
+    location = json.decode(location) if location else {}
+    time_format = H12 if config.get(P_USE_12H) else H24
+    blink_time = config.get(P_BLINK_TIME)
+
+    timezone = location.get(
+        "timezone",
+        config.get("$tz", DEFAULT_TIMEZONE),
+    )
+    now = time.now().in_location(timezone)
+    now_date = now.format("Mon 2 Jan 2006")
+
+    # this is a rough approximation
+    # actual seasons vary by location and astronomy
+    # we just treat this as close enough
+    season = int(now.format("1-02").replace("-", ""))
+    if season >= 320 and season < 621:
+        season = 1
+    elif season >= 621 and season < 923:
+        season = 2
+    elif season >= 923 and season < 1221:
+        season = 3
+    else:
+        season = 0
+
+    phase = int(now.format("15"))
```

Comment:
nit: you could use the new sunrise/sunset timer!

Reply:
I actually *don't* want it to go by sunrise and sunset.  I should probably make this configurable.  I have time to think about it since the color picker is still up in the air.

Reply:
I made this configurable.